### ISSUE TITLE
Add aria-disabled to buttons

### ISSUE THREAD

**Issue by bernbecht** [Shopify/polaris-react-archive#5228]
## 🙏 Feature request summary
We wonder why [`<UnstyledButton/>`](https://github.com/Shopify/polaris-react/blob/main/src/components/UnstyledButton/UnstyledButton.tsx#L77) doesn't have `aria-disabled` prop.

## 🧠 Rationale
We have the following use case:
-  `<UnstyledButton/>` button is disabled
- user should be able to trigger a tooltip (hover & focus) explaining why the button is disabled
- Voice Over needs to read the _name_, _state_, _role_ and ideally the tooltip's content when user focus on the button. The voice over would read something like _"button label, button, dimmed, tooltip content"_

<img src='https://screenshot.click/01-51-3g9tq-52fdn.gif' width='500px' /> 

Today this use case isn't possible because:
1) Disabled buttons are not focusable
2) Using `Tooltip`+ disabled `Button` makes Voice Over to read the wrong element (tooltip is the firstFocusableNode). [Sandbox](https://codesandbox.io/s/fervent-https-ecdvpu?file=/App.js:579-691)
<img src='https://screenshot.click/03-58-fdqg8-6pd6a.gif' width='500px' /> 

By adding `aria-disabled` prop to buttons, developers can make disabled buttons accessible. In this way, all users can find and understand why a button is disable and act from there.

**Comment by ghost** [comment]
👋 Thanks for opening your first issue. A contributor should give feedback soon. If you haven’t already, please check out the [contributing guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md).

**Comment by luke-underwood** [comment]
Looks like unstyled button now uses `aria-disabled`. Should this be closed?

https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/UnstyledButton/UnstyledButton.tsx#L85

**Comment by bernbecht** [comment]
Yes! Thank you for flagging it here
